src/ckeditor.jsx

```diff
@@ -51,60 +94,97 @@ export default class CKEditor extends React.Component {
 	}
 
 	_initializeEditor() {
-		this.props.editor
-			.create( this.domContainer.current, this._getConfig() )
-			.then( editor => {
-				this.editor = editor;
-
-				if ( 'disabled' in this.props ) {
-					editor.isReadOnly = this.props.disabled;
-				}
-
-				if ( this.props.onInit ) {
-					this.props.onInit( editor );
-				}
-
-				const modelDocument = editor.model.document;
-				const viewDocument = editor.editing.view.document;
-
-				modelDocument.on( 'change:data', event => {
-					/* istanbul ignore else */
-					if ( this.props.onChange ) {
-						this.props.onChange( event, editor );
+		const creator = ( el, config ) => {
```

Comment:
What about moving the creator function outside the component as a factory function?

```js
function creatorFactory( element, editorConfig, componentProperties ) {
    return ( el, config ) { /* ... */ };
}
```

Thanks to that, the `_initializeEditor()` body will be a little cleaner.